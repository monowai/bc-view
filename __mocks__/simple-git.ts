jest.mock("simple-git")
const mockGit = {
  __esModule: true,
  default: (): { revparse: jest.Mock } => ({
    revparse: jest
      .fn()
      .mockResolvedValueOnce("mock-branch")
      .mockResolvedValueOnce("mock-commit"),
  }),
}
export default jest.fn(() => mockGit)
