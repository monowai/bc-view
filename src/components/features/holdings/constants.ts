// Constants for GrandTotal component structure and layout

export const GRANDTOTAL_LAYOUT = {
  // Cell structure constants
  TOTAL_CELLS: 14,           // 1 label + 1 spacer + 12 data cells
  LABEL_CELL_COUNT: 1,       // Value title cell
  SPACER_CELL_COUNT: 1,      // Spacer cell that skips Price column
  DATA_CELL_COUNT: 12,       // Actual data cells

  // Slice positions for test data extraction
  DATA_CELLS_SLICE_START: 2, // Position after label and spacer cells

  // Column spans
  DEFAULT_COLSPAN: 1,
  COST_VALUE_COLSPAN: 2,     // Cost value spans quantity + cost columns

  // Cell positions (after slice)
  CHANGE_POSITION: 0,        // Change column (empty)
  GAIN_ON_DAY_POSITION: 1,   // Gain on day value
  COST_VALUE_POSITION: 2,    // Cost value (quantity column)
  COST_EMPTY_POSITION: 3,    // Cost column (empty)
  MARKET_VALUE_POSITION: 4,  // Market value
  DIVIDENDS_POSITION: 5,     // Dividends
  UNREALISED_GAIN_POSITION: 6, // Unrealised gain
  REALISED_GAIN_POSITION: 7, // Realised gain
  IRR_POSITION: 8,           // IRR percentage
  ALPHA_POSITION: 9,         // Alpha column (spacer)
  WEIGHT_POSITION: 10,       // Weight percentage
  TOTAL_GAIN_POSITION: 11,   // Total gain
} as const

export const RESPONSIVE_BREAKPOINTS = {
  // Viewport widths for testing
  MOBILE_WIDTH: 375,         // < 768px
  TABLET_WIDTH: 768,         // 768px - 1199px
  DESKTOP_WIDTH: 1400,       // 1200px+
} as const

export const MULTIPLIERS = {
  PERCENTAGE: 100,           // For IRR and Weight display (0.15 -> 15.00%)
  DEFAULT: 1,                // For currency values
} as const

export const TEST_VALUES = {
  // Expected values from mock data
  GAIN_ON_DAY: '72.76',
  COST_VALUE: '8,150.65',
  MARKET_VALUE: '12,643.74',
  DIVIDENDS: '299.02',
  UNREALISED_GAIN: '3,503.85',
  REALISED_GAIN: '481.44',
  IRR: '15.00',              // 0.15 * 100 = 15.00
  WEIGHT: '100.00%',         // 1.0 * 100 = 100.00%
  TOTAL_GAIN: '4,284.31',
  VALUE_TITLE: 'Value in PORTFOLIO',
} as const