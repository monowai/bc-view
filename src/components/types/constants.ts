// Enums can't be exported from types in time to compilation to resolve

export enum GroupBy {
  MARKET_CURRENCY = "asset.market.currency.code",
  MARKET = "asset.market.code",
  ASSET_CLASS = "asset.assetCategory.name",
}

// Enum is pointer to a collection of values that index the holding contract
export enum ValueIn {
  PORTFOLIO = "PORTFOLIO",
  BASE = "BASE",
  TRADE = "TRADE",
}
