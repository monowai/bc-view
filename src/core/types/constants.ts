// export type ValueIn = "TRADE" | "BASE" | "PORTFOLIO";

import { ValuationOption } from "./beancounter";
import { t } from "i18next";

// Enum is pointer to a collection of values in the holding contract
export enum ValueIn {
  TRADE = "TRADE",
  BASE = "BASE",
  PORTFOLIO = "PORTFOLIO",
}

export function valuationOptions(): ValuationOption[] {
  return [
    { value: ValueIn.PORTFOLIO, label: "Portfolio" },
    { value: ValueIn.BASE, label: "Base" },
    { value: ValueIn.TRADE, label: "Trade" },
  ];
}

export const __new__ = "new";
