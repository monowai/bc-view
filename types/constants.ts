// Enums can't be exported from types in time to compilation to resolve
