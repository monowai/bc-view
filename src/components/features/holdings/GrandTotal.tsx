import { HoldingsInCurrency } from "types/beancounter"
import React, { ReactElement } from "react"
import { FormatValue } from "@components/ui/MoneyUtils"
import { useTranslation } from "next-i18next"
import { headers, HEADER_INDICES } from "./Header"
import { GRANDTOTAL_LAYOUT } from "./constants"

// Totals all subtotal values.
export default function GrandTotal({
  holdings,
  valueIn,
}: HoldingsInCurrency): ReactElement {
  const { t, ready } = useTranslation("common")
  if (!ready) return <div />
  if (!holdings.viewTotals) return <div />

  // GrandTotal data: skips first column (Price), maps to headers[1-12]
  const data: { value: number | string | null; colSpan: number; multiplier?: number }[] = [
    { value: null, colSpan: 1 },                               // Maps to CHANGE% (headers[1])
    { value: holdings.viewTotals.gainOnDay || 0, colSpan: 1 }, // Maps to GAIN_ON_DAY (headers[2])
    { value: holdings.viewTotals.costValue, colSpan: 1 },      // Maps to QUANTITY (headers[3]) - costValue only in quantity column
    { value: null, colSpan: 1 },                              // Maps to COST (headers[4]) - empty since cost column gets the costValue
    { value: holdings.viewTotals.marketValue, colSpan: 1 },    // Maps to MARKET_VALUE (headers[5])
    { value: holdings.viewTotals.dividends, colSpan: 1 },      // Maps to DIVIDENDS (headers[6])
    { value: holdings.viewTotals.unrealisedGain, colSpan: 1 }, // Maps to UNREALISED_GAIN (headers[7])
    { value: holdings.viewTotals.realisedGain, colSpan: 1 },   // Maps to REALISED_GAIN (headers[8])
    { value: holdings.viewTotals.irr, colSpan: 1, multiplier: 100 }, // Maps to IRR (headers[9])
    { value: null, colSpan: 1 },                               // Maps to ALPHA (headers[10])
    { value: holdings.viewTotals.weight, colSpan: 1, multiplier: 100 }, // Maps to WEIGHT (headers[11])
    { value: holdings.viewTotals.totalGain, colSpan: 1 },      // Maps to TOTAL_GAIN (headers[12])
  ]

  return (
    <tbody className="grand-totals" key={holdings.portfolio.code + "totals"}>
      <tr>
        <td colSpan={GRANDTOTAL_LAYOUT.TOTAL_CELLS} className="border-t-2 border-gray-600" />
      </tr>
      <tr
        key={valueIn}
        className="holding-footer text-sm bg-gray-100 border-b-2 border-gray-600 hover:!bg-slate-200 transition-colors duration-200"
      >
        <td className="px-1 py-1 md:px-2 xl:px-4 text-xs md:text-sm font-medium text-right">
          <div>{t("holdings.valueTitle", { valueIn })}</div>
        </td>
        {/* Skip only Price column (colSpan=1) - Change column should be visible */}
        <td colSpan={1} />
        {data.map((item, index) => {
          // Explicit mapping for each data position to ensure correct alignment
          let headerIndex
          switch(index) {
            case 0: headerIndex = HEADER_INDICES.CHANGE; break           // change (null)
            case 1: headerIndex = HEADER_INDICES.GAIN_ON_DAY; break      // gainOnDay
            case 2: headerIndex = HEADER_INDICES.QUANTITY; break         // costValue in quantity column
            case 3: headerIndex = HEADER_INDICES.COST; break             // cost column (empty)
            case 4: headerIndex = HEADER_INDICES.MARKET_VALUE; break     // marketValue
            case 5: headerIndex = HEADER_INDICES.DIVIDENDS; break        // dividends
            case 6: headerIndex = HEADER_INDICES.UNREALISED_GAIN; break  // unrealisedGain
            case 7: headerIndex = HEADER_INDICES.REALISED_GAIN; break    // realisedGain
            case 8: headerIndex = HEADER_INDICES.IRR; break              // irr
            case 9: headerIndex = HEADER_INDICES.ALPHA; break            // alpha (null)
            case 10: headerIndex = HEADER_INDICES.WEIGHT; break          // weight
            case 11: headerIndex = HEADER_INDICES.TOTAL_GAIN; break      // totalGain
            default: headerIndex = index + 1
          }

          const header = headers[headerIndex]

          // All columns follow header rules exactly
          let visibility
          if (header?.mobile) {
            visibility = "" // Visible on all screens if mobile: true
          } else if (header?.medium) {
            visibility = "hidden md:table-cell"
          } else {
            visibility = "hidden xl:table-cell"
          }

          // Get alignment from headers array
          const alignment = header?.align || "right"

          // Determine color for gainOnDay column
          let colorClass = ""
          if (headerIndex === HEADER_INDICES.GAIN_ON_DAY && typeof item.value === "number") {
            if (item.value < 0) {
              colorClass = "text-red-500"
            } else if (item.value > 0) {
              colorClass = "text-green-500"
            }
          }

          return (
            <td
              key={index}
              colSpan={item.colSpan}
              className={`px-1 py-1 md:px-2 xl:px-4 text-xs md:text-sm font-medium text-${alignment} ${visibility} ${colorClass}`}
            >
              {item.value !== null && item.value !== "" ? (
                <>
                  <FormatValue
                    value={item.value}
                    defaultValue=""
                    multiplier={headerIndex === HEADER_INDICES.IRR || headerIndex === HEADER_INDICES.WEIGHT ? 100 : 1}
                  />
                  {headerIndex === HEADER_INDICES.WEIGHT && "%"}
                  {/* Remove % signs from IRR for cleaner appearance */}
                </>
              ) : (
                typeof item.value === "string" ? item.value : (headerIndex === HEADER_INDICES.GAIN_ON_DAY ? "0" : "")
              )}
            </td>
          )
        })}
      </tr>
    </tbody>
  )
}
