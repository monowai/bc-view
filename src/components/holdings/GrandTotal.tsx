import { HoldingsInCurrency } from "@components/types/beancounter";
import React, { ReactElement } from "react";
import { FormatValue } from "@components/MoneyUtils";
import { useTranslation } from "next-i18next";

// Totals all subtotal values.
export default function GrandTotal({
  holdings,
  valueIn,
}: HoldingsInCurrency): ReactElement {
  const { t, ready } = useTranslation("common");
  if (!ready) return <div />;
  if (!holdings.viewTotals) return <div />;

  const data = [
    { value: holdings.viewTotals.gainOnDay, colSpan: 1 },
    { value: holdings.viewTotals.costValue, colSpan: 2 },
    { value: holdings.viewTotals.marketValue, colSpan: 1 },
    { value: holdings.viewTotals.unrealisedGain, colSpan: 1 },
    { value: holdings.viewTotals.weight, colSpan: 1 },
    { value: holdings.viewTotals.dividends, colSpan: 1 },
    { value: holdings.viewTotals.realisedGain, colSpan: 1 },
    { value: holdings.viewTotals.totalGain, colSpan: 2 },
  ];

  return (
    <tbody className={"totals-row"} key={holdings.portfolio.code + "totals"}>
      <tr key={valueIn}>
        <td colSpan={3} align={"right"}>
          <div>{t("holdings.valueTitle", { valueIn })}</div>
        </td>
        {data.map((item, index) => (
          <td key={index} colSpan={item.colSpan} align={"right"}>
            <FormatValue
              value={item.value}
              defaultValue="-"
              multiplier={index === 4 ? 100 : 1}
            />
          </td>
        ))}
      </tr>
    </tbody>
  );
}
