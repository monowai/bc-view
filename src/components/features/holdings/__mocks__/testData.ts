// Shared mock holdings data for GrandTotal tests
import { ValueIn } from '@components/features/holdings/GroupByOptions'

export const mockHoldings = {
  holdingGroups: {},
  currency: { code: 'USD', name: 'US Dollar', symbol: '$' },
  portfolio: {
    id: 'test-portfolio',
    code: 'TEST',
    name: 'Test Portfolio',
    currency: { code: 'USD', name: 'US Dollar', symbol: '$' },
    base: { code: 'USD', name: 'US Dollar', symbol: '$' },
    marketValue: 12643.74,
    irr: 0.15
  },
  valueIn: ValueIn.PORTFOLIO,
  viewTotals: {
    gainOnDay: 72.76,
    costValue: 8150.65,
    marketValue: 12643.74,
    dividends: 299.02,
    unrealisedGain: 3503.85,
    realisedGain: 481.44,
    irr: 0.15,
    weight: 1.0,
    totalGain: 4284.31,
    currency: { code: 'USD', name: 'US Dollar', symbol: '$' },
    fees: 0,
    tax: 0,
    cash: 0,
    purchases: 8150.65,
    sales: 0,
    costBasis: 8150.65,
    averageCost: 45.28,
    priceData: { change: 0, close: 70.25, previousClose: 70.25, changePercent: 0, priceDate: '2023-01-01' },
    valueIn: { code: 'USD', name: 'US Dollar', symbol: '$' },
    roi: 0.55,
    [ValueIn.PORTFOLIO]: {} as any
  },
  totals: {
    marketValue: 12643.74,
    purchases: 8150.65,
    sales: 0,
    cash: 0,
    income: 299.02,
    gain: 4284.31,
    irr: 0.15,
    currency: { code: 'USD', name: 'US Dollar', symbol: '$' }
  }
}

export const mockValueIn = ValueIn.PORTFOLIO

// Shared mock implementation for next-i18next
export const mockUseTranslation = (): { t: (key: string, options?: any) => string; ready: boolean } => ({
  t: (key: string, options?: any) => {
    if (key === 'holdings.valueTitle') return `Value in ${options?.valueIn || 'USD'}`
    return key
  },
  ready: true
})