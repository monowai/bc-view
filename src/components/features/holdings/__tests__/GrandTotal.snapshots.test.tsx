import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import GrandTotal from '../GrandTotal'
import { mockHoldings, mockValueIn, mockUseTranslation } from '../__mocks__/testData'
import { GRANDTOTAL_LAYOUT, RESPONSIVE_BREAKPOINTS } from '../constants'

// Mock next-i18next
jest.mock('next-i18next', () => ({
  useTranslation: () => mockUseTranslation()
}))

describe('GrandTotal Snapshot Tests', () => {
  afterEach(() => {
    // Reset viewport after each test
    Object.defineProperty(window, 'innerWidth', {
      writable: true,
      configurable: true,
      value: 1024
    })
  })

  it('matches desktop snapshot', () => {
    // Mock desktop viewport (1200px+)
    Object.defineProperty(window, 'innerWidth', {
      writable: true,
      configurable: true,
      value: RESPONSIVE_BREAKPOINTS.DESKTOP_WIDTH
    })

    const { container } = render(
      <table>
        <GrandTotal holdings={mockHoldings} valueIn={mockValueIn} />
      </table>
    )

    expect(container.firstChild).toMatchSnapshot('desktop-layout')
  })

  it('matches tablet snapshot', () => {
    // Mock tablet viewport (768px - 1199px)
    Object.defineProperty(window, 'innerWidth', {
      writable: true,
      configurable: true,
      value: RESPONSIVE_BREAKPOINTS.TABLET_WIDTH
    })

    const { container } = render(
      <table>
        <GrandTotal holdings={mockHoldings} valueIn={mockValueIn} />
      </table>
    )

    expect(container.firstChild).toMatchSnapshot('tablet-layout')
  })

  it('matches mobile snapshot', () => {
    // Mock mobile viewport (< 768px)
    Object.defineProperty(window, 'innerWidth', {
      writable: true,
      configurable: true,
      value: RESPONSIVE_BREAKPOINTS.MOBILE_WIDTH
    })

    const { container } = render(
      <table>
        <GrandTotal holdings={mockHoldings} valueIn={mockValueIn} />
      </table>
    )

    expect(container.firstChild).toMatchSnapshot('mobile-layout')
  })

  describe('Regression Protection', () => {
    it('preserves desktop column structure (all 13 columns present)', () => {
      Object.defineProperty(window, 'innerWidth', {
        writable: true,
        configurable: true,
        value: 1400
      })

      const { container } = render(
        <table>
          <GrandTotal holdings={mockHoldings} valueIn={mockValueIn} />
        </table>
      )

      const dataRow = container.querySelector('tbody tr:last-child')
      const cells = dataRow?.querySelectorAll('td')

      // Should have all cells: label + spacer + data cells
      expect(cells).toHaveLength(GRANDTOTAL_LAYOUT.TOTAL_CELLS)

      // All data cells should be visible on desktop
      const dataCells = Array.from(cells!).slice(GRANDTOTAL_LAYOUT.DATA_CELLS_SLICE_START)

      // On desktop, all columns should be visible - none should have 'hidden' without a responsive suffix
      // Mobile-only hidden: 'hidden md:table-cell' or 'hidden xl:table-cell' (both visible on desktop)
      // Desktop-only hidden: would be 'hidden' with no suffix (but all columns should be visible on desktop)
      const hiddenOnDesktop = dataCells.filter(cell => {
        const classes = cell.className
        // Hidden on desktop if it has 'hidden' but no responsive suffix that makes it visible on desktop
        return classes.includes('hidden') && !classes.includes('md:table-cell') && !classes.includes('xl:table-cell')
      })

      // Should have 0 columns hidden on desktop (all 11 data columns visible)
      expect(hiddenOnDesktop).toHaveLength(0)

      // Verify we have all data cells
      expect(dataCells).toHaveLength(GRANDTOTAL_LAYOUT.DATA_CELL_COUNT)
    })

    it('preserves tablet column structure (9 visible columns)', () => {
      Object.defineProperty(window, 'innerWidth', {
        writable: true,
        configurable: true,
        value: 768
      })

      const { container } = render(
        <table>
          <GrandTotal holdings={mockHoldings} valueIn={mockValueIn} />
        </table>
      )

      const dataRow = container.querySelector('tbody tr:last-child')
      const cells = dataRow?.querySelectorAll('td')
      const dataCells = Array.from(cells!).slice(2)

      // Count visible columns on tablet (excluding hidden columns)
      const visibleColumns = dataCells.filter(cell => {
        const classes = cell.className
        return !classes.includes('hidden') || classes.includes('md:table-cell')
      })

      // Should have 10 visible columns on tablet
      expect(visibleColumns).toHaveLength(10)

      // Verify key columns are in correct positions
      expect(dataCells[0]).toHaveTextContent('') // Change (empty)
      expect(dataCells[1]).toHaveTextContent('72.76') // gainOnDay
      expect(dataCells[4]).toHaveTextContent(/12,?643\.74/) // marketValue
      expect(dataCells[11]).toHaveTextContent(/4,?284\.31/) // totalGain
    })
  })
})