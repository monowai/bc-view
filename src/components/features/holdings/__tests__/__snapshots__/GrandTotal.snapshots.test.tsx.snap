// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`GrandTotal Snapshot Tests matches desktop snapshot: desktop-layout 1`] = `
<table>
  <tbody
    class="grand-totals"
  >
    <tr>
      <td
        class="border-t-2 border-gray-600"
        colspan="14"
      />
    </tr>
    <tr
      class="holding-footer text-sm bg-gray-100 border-b-2 border-gray-600 hover:!bg-slate-200 transition-colors duration-200"
    >
      <td
        class="px-1 py-1 md:px-2 xl:px-4 text-xs md:text-sm font-medium text-right"
      >
        <div>
          Value in PORTFOLIO
        </div>
      </td>
      <td
        colspan="1"
      />
      <td
        class="px-1 py-1 md:px-2 xl:px-4 text-xs md:text-sm font-medium text-right  "
        colspan="1"
      />
      <td
        class="px-1 py-1 md:px-2 xl:px-4 text-xs md:text-sm font-medium text-right  text-green-500"
        colspan="1"
      >
        <span>
          72.76
        </span>
      </td>
      <td
        class="px-1 py-1 md:px-2 xl:px-4 text-xs md:text-sm font-medium text-right  "
        colspan="1"
      >
        <span>
          8,150.65
        </span>
      </td>
      <td
        class="px-1 py-1 md:px-2 xl:px-4 text-xs md:text-sm font-medium text-right hidden md:table-cell "
        colspan="1"
      />
      <td
        class="px-1 py-1 md:px-2 xl:px-4 text-xs md:text-sm font-medium text-right  "
        colspan="1"
      >
        <span>
          12,643.74
        </span>
      </td>
      <td
        class="px-1 py-1 md:px-2 xl:px-4 text-xs md:text-sm font-medium text-right hidden md:table-cell "
        colspan="1"
      >
        <span>
          299.02
        </span>
      </td>
      <td
        class="px-1 py-1 md:px-2 xl:px-4 text-xs md:text-sm font-medium text-right hidden md:table-cell "
        colspan="1"
      >
        <span>
          3,503.85
        </span>
      </td>
      <td
        class="px-1 py-1 md:px-2 xl:px-4 text-xs md:text-sm font-medium text-right hidden md:table-cell "
        colspan="1"
      >
        <span>
          481.44
        </span>
      </td>
      <td
        class="px-1 py-1 md:px-2 xl:px-4 text-xs md:text-sm font-medium text-right  "
        colspan="1"
      >
        <span>
          15.00
        </span>
      </td>
      <td
        class="px-1 py-1 md:px-2 xl:px-4 text-xs md:text-sm font-medium text-center hidden xl:table-cell "
        colspan="1"
      />
      <td
        class="px-1 py-1 md:px-2 xl:px-4 text-xs md:text-sm font-medium text-right hidden xl:table-cell "
        colspan="1"
      >
        <span>
          100.00
        </span>
        %
      </td>
      <td
        class="px-1 py-1 md:px-2 xl:px-4 text-xs md:text-sm font-medium text-right  "
        colspan="1"
      >
        <span>
          4,284.31
        </span>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`GrandTotal Snapshot Tests matches mobile snapshot: mobile-layout 1`] = `
<table>
  <tbody
    class="grand-totals"
  >
    <tr>
      <td
        class="border-t-2 border-gray-600"
        colspan="14"
      />
    </tr>
    <tr
      class="holding-footer text-sm bg-gray-100 border-b-2 border-gray-600 hover:!bg-slate-200 transition-colors duration-200"
    >
      <td
        class="px-1 py-1 md:px-2 xl:px-4 text-xs md:text-sm font-medium text-right"
      >
        <div>
          Value in PORTFOLIO
        </div>
      </td>
      <td
        colspan="1"
      />
      <td
        class="px-1 py-1 md:px-2 xl:px-4 text-xs md:text-sm font-medium text-right  "
        colspan="1"
      />
      <td
        class="px-1 py-1 md:px-2 xl:px-4 text-xs md:text-sm font-medium text-right  text-green-500"
        colspan="1"
      >
        <span>
          72.76
        </span>
      </td>
      <td
        class="px-1 py-1 md:px-2 xl:px-4 text-xs md:text-sm font-medium text-right  "
        colspan="1"
      >
        <span>
          8,150.65
        </span>
      </td>
      <td
        class="px-1 py-1 md:px-2 xl:px-4 text-xs md:text-sm font-medium text-right hidden md:table-cell "
        colspan="1"
      />
      <td
        class="px-1 py-1 md:px-2 xl:px-4 text-xs md:text-sm font-medium text-right  "
        colspan="1"
      >
        <span>
          12,643.74
        </span>
      </td>
      <td
        class="px-1 py-1 md:px-2 xl:px-4 text-xs md:text-sm font-medium text-right hidden md:table-cell "
        colspan="1"
      >
        <span>
          299.02
        </span>
      </td>
      <td
        class="px-1 py-1 md:px-2 xl:px-4 text-xs md:text-sm font-medium text-right hidden md:table-cell "
        colspan="1"
      >
        <span>
          3,503.85
        </span>
      </td>
      <td
        class="px-1 py-1 md:px-2 xl:px-4 text-xs md:text-sm font-medium text-right hidden md:table-cell "
        colspan="1"
      >
        <span>
          481.44
        </span>
      </td>
      <td
        class="px-1 py-1 md:px-2 xl:px-4 text-xs md:text-sm font-medium text-right  "
        colspan="1"
      >
        <span>
          15.00
        </span>
      </td>
      <td
        class="px-1 py-1 md:px-2 xl:px-4 text-xs md:text-sm font-medium text-center hidden xl:table-cell "
        colspan="1"
      />
      <td
        class="px-1 py-1 md:px-2 xl:px-4 text-xs md:text-sm font-medium text-right hidden xl:table-cell "
        colspan="1"
      >
        <span>
          100.00
        </span>
        %
      </td>
      <td
        class="px-1 py-1 md:px-2 xl:px-4 text-xs md:text-sm font-medium text-right  "
        colspan="1"
      >
        <span>
          4,284.31
        </span>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`GrandTotal Snapshot Tests matches tablet snapshot: tablet-layout 1`] = `
<table>
  <tbody
    class="grand-totals"
  >
    <tr>
      <td
        class="border-t-2 border-gray-600"
        colspan="14"
      />
    </tr>
    <tr
      class="holding-footer text-sm bg-gray-100 border-b-2 border-gray-600 hover:!bg-slate-200 transition-colors duration-200"
    >
      <td
        class="px-1 py-1 md:px-2 xl:px-4 text-xs md:text-sm font-medium text-right"
      >
        <div>
          Value in PORTFOLIO
        </div>
      </td>
      <td
        colspan="1"
      />
      <td
        class="px-1 py-1 md:px-2 xl:px-4 text-xs md:text-sm font-medium text-right  "
        colspan="1"
      />
      <td
        class="px-1 py-1 md:px-2 xl:px-4 text-xs md:text-sm font-medium text-right  text-green-500"
        colspan="1"
      >
        <span>
          72.76
        </span>
      </td>
      <td
        class="px-1 py-1 md:px-2 xl:px-4 text-xs md:text-sm font-medium text-right  "
        colspan="1"
      >
        <span>
          8,150.65
        </span>
      </td>
      <td
        class="px-1 py-1 md:px-2 xl:px-4 text-xs md:text-sm font-medium text-right hidden md:table-cell "
        colspan="1"
      />
      <td
        class="px-1 py-1 md:px-2 xl:px-4 text-xs md:text-sm font-medium text-right  "
        colspan="1"
      >
        <span>
          12,643.74
        </span>
      </td>
      <td
        class="px-1 py-1 md:px-2 xl:px-4 text-xs md:text-sm font-medium text-right hidden md:table-cell "
        colspan="1"
      >
        <span>
          299.02
        </span>
      </td>
      <td
        class="px-1 py-1 md:px-2 xl:px-4 text-xs md:text-sm font-medium text-right hidden md:table-cell "
        colspan="1"
      >
        <span>
          3,503.85
        </span>
      </td>
      <td
        class="px-1 py-1 md:px-2 xl:px-4 text-xs md:text-sm font-medium text-right hidden md:table-cell "
        colspan="1"
      >
        <span>
          481.44
        </span>
      </td>
      <td
        class="px-1 py-1 md:px-2 xl:px-4 text-xs md:text-sm font-medium text-right  "
        colspan="1"
      >
        <span>
          15.00
        </span>
      </td>
      <td
        class="px-1 py-1 md:px-2 xl:px-4 text-xs md:text-sm font-medium text-center hidden xl:table-cell "
        colspan="1"
      />
      <td
        class="px-1 py-1 md:px-2 xl:px-4 text-xs md:text-sm font-medium text-right hidden xl:table-cell "
        colspan="1"
      >
        <span>
          100.00
        </span>
        %
      </td>
      <td
        class="px-1 py-1 md:px-2 xl:px-4 text-xs md:text-sm font-medium text-right  "
        colspan="1"
      >
        <span>
          4,284.31
        </span>
      </td>
    </tr>
  </tbody>
</table>
`;
